### PR TITLE
Update to Agents v6

### LINKED ISSUE #50: Rework microbe type definitions using `@agent` macro
To be done after https://github.com/JuliaDynamics/Agents.jl/pull/843 is merged and released.

### PR DESCRIPTION
**Pull request by mastrof**
- Use `SVector`s instead of `NTuple`s
- Define microbes with the `@agent` keyword (Closes #50)
- General API improvements

Currently tracking Agents#main because still unreleased

### CODE REVIEW

File: src/chemotaxis/brown-berg.jl [hunk 1]
```diff
@@ -14,40 +14,17 @@ Default parameters:
 - `receptor_binding_constant = 100` μM
 - `memory = 1` s
 """
-mutable struct BrownBerg{D} <: AbstractMicrobe{D}
-    id::Int
-    pos::NTuple{D,Float64}
-    motility::AbstractMotility
-    vel::NTuple{D,Float64}
+@agent BrownBerg{D} ContinuousAgent{D,Float64} where D AbstractMicrobe{D} begin
```
Comment (Tortar) [inline src/chemotaxis/brown-berg.jl:17]:
```suggestion
@agent BrownBerg{D} ContinuousAgent{D,Float64} AbstractMicrobe{D} begin
```

Actual syntax ;)

Comment (Tortar) [reply src/chemotaxis/brown-berg.jl]: See your commits, I don't why I thought the one I suggested was the actual syntax :D

Comment (mastrof) [reply src/chemotaxis/brown-berg.jl]: I guess with all the changes that have been happening in the last 2 months, some confusion is allowed

**Comment by mastrof** [comment]
Need to merge even though agents v6 is not out yet

**Pull request merged**